### DISCUSSION TITLE
Embedding Music Assistant Lovelace View?

### DISCUSSION THREAD

**Discussion by StifflersMoM69** [music-assistant/support#1693] [Q&A] [answered]
Hi there!

is it possible to show Music Assistant instead in the menu of the left side on the top in a dashboard?

**Comment by OzGav** [comment]
https://community.home-assistant.io/t/how-to-embed-a-webpage-in-a-dashboard-view/400660/3

**Reply by StifflersMoM69** [reply]
Thanks for the reply, but when i do use the ebpageview, also the sidepanel is loded too (homeassistant). just want to display all within Music Assistant (d5369777_music_assistant_beta/ingress)

**Reply by OzGav** [reply]
Maybe try exposing the web server port and see if using that works instead?

**Reply by pressurejets** [reply]
ingress error 401 ?? how can i get around it.

**Reply by Kree-Gore** [reply]
Same Question here. Don´t get the exposing of a port...can´t find a good tutorial. In Grafana it explicitly asks for port-exposure...

**Reply by OzGav** [reply]
Not sure what the question is. Have you exposed the web server port in the MA settings?

**Reply by Kree-Gore** [reply]
No, do not know how to. Searchin for hours...

**Reply by OzGav** [reply]
SETTINGS>> CORE>>WEBSERVER

**Reply by Kree-Gore** [reply]
No Core in the settings or webserver in Home Assistant OS with current version:

![image](https://github.com/music-assistant/hass-music-assistant/assets/65222903/e01d7c22-1069-4762-9a94-347e666669d7)

**Reply by OzGav** [reply]
You need the MA settings. Go to the addon.

**Reply by Kree-Gore** [reply]
Okay, go on please:

![image](https://github.com/music-assistant/hass-music-assistant/assets/65222903/0dd82e29-2284-4f20-9b1e-c89f733f0880)

**Reply by OzGav** [reply]
Actually go into MA. You will then see a cog on the left side menu buttons

**Reply by Kree-Gore** [reply]
Found it, very nice, restarted HA and now It should be working, thank you. I seem to have a login problem with another user, before MASS is started (after being idle?) but that is another problem. FYI: Another workaround would be to link "within HA" so only: "/api/hassio_ingress/xyz/#/search" without anything before.

![image](https://github.com/music-assistant/hass-music-assistant/assets/65222903/2521d425-fec0-4d93-b0de-7987b02dfd94)

**Reply by OzGav** [reply]
Yes since you are using the addon, for security reasons, we recommend you don’t expose the port. Just click on the addon option (In the INFORMATIONEN view) to show in sidebar and access via ingress (as you have discovered)

**Reply by Kree-Gore** [reply]
Hm, but the direct link: http://192.168.188.89:8095/d5369777_music_assistant_beta/ingress (or with homeassistant.local) does not work (404)....On my tablet with kiosk browser I still get a 401 with the other method. It is a different user and a different device (fully kiosk) browser. As soon as I klick into the addon there and then try the popup link, it is working. So I thought with the direct link it should work, but now I get a 404 when using the port. So it seems that:

1. MASS needs the cookie from the device (blocked popups in kiosk and internal navigation does not seem to work (as in navigatefor once to the addon an go back)
2. The direct link with the port still causes problems, besides being exposed. The linkage should be correct (same with grafana, working well, no disruptions)

**Reply by OzGav** [reply, marked as answer]
Sorry why aren’t you using 
192.168.188.89:8123/hassio/ingress/d5369777_music_assistant_beta
Or just
192.168.188.89:8095

if you have some complex network setup happening then that will be the problem. Everything needs to be in the same subnet.

**Reply by Kree-Gore** [reply]
Ah, you are god =) without the hassio etc. it works. simply http://homeassistant.local:8095/#/search directly for search option, PERFECT, MANY MANY THANKS!!!:

![image](https://github.com/music-assistant/hass-music-assistant/assets/65222903/070c3bd1-8f82-4bcf-843f-86544038a8f4)

**Reply by StifflersMoM69** [reply]
That doesnt seems to work with https

**Reply by OzGav** [reply]
Have a look at the FAQ regarding the use of SSL internally. Long story short it is not recommended and is unnecessary in a home internal network.
